server/src/code-runner/docker-runner.ts

```diff
@@ -48,40 +74,46 @@ export class DockerRunner implements CodeRunner {
 
     const mounts = flatMap(configuration.mountPoints, mp => ['-v', `${mp.source}:${mp.destination}:ro`]);
 
-    return this.config
-      .spawn(this.config.dockerExecutable, [
-        'create',
-        '--cap-drop=ALL',
-        `--kernel-memory=${this.config.maxKernelMemoryKb}k`,
-        '--security-opt=no-new-privileges',
-        '-t',
-        '--read-only',
-        '--tmpfs',
-        `/run:rw,size=${this.config.runPartitionSize},mode=${this.config.runPartitionMode}`,
-        '--tmpfs',
-        `/tmp:rw,size=${this.config.tmpPartitionSize},mode=${this.config.tmpPartitionMode}`,
-        ...mounts,
-        '-v',
-        `${tmpExecutionDir}:/lab:ro`,
-        `--name`,
-        invocation.id,
-        configuration.imageWithDigest,
-        '/bin/bash'
-      ])
+    return new UserResolver().resolve(invocation)
       .pipe(
+        map(user => {
+          if (!environment.writeable) {
+            return ['--read-only'];
+          }
+
+          const plan = PlanCredits.get(user.plan.plan_id);
```

Comment:
I feel that the `DockerRunner`  is intended to be low level enough to not know about the user at all. So rather than checking the plan here, I think the `maxWriteableContainerSizeInGb` should rather be something that ends up on the passed in `config` (`InternalLabConfiguration`)

Reply:
Thanks to your explanation above, this is now done in the transformer.